pkg/storegateway/bucket.go

```diff
@@ -615,7 +615,80 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		readers = newChunkReaders(chunkReaders)
 	}
```

Comment:
General comment on `Series`: it was already huge and now we're making it huge-r 😄 Is there any way we can refactor it to make it smaller and easier to follow? (Might be something for a follow-up PR.)